src/main/java/reactor/netty/transport/ProxyProvider.java

```diff
@@ -324,6 +340,65 @@ public ProxyProvider build() {
 		}
 	}
 
+	static final class RegexShouldProxyPredicate implements Predicate<SocketAddress> {
+
+		private final String regex;
+		private Pattern pattern;
+
+		/**
+		 * Create a {@link RegexShouldProxyPredicate} based off the provided regular expression.
+		 *
+		 * @param regex The string regular expression
+		 * @return a predicate whether we should direct to proxy
+		 */
+		public static RegexShouldProxyPredicate fromRegularExpression(String regex) {
+			return new RegexShouldProxyPredicate(regex);
+		}
+
+		/**
+		 * Creates a {@link RegexShouldProxyPredicate} based off the provided pattern with possible wildcards as
+		 * described in https://docs.oracle.com/javase/7/docs/api/java/net/doc-files/net-properties.html
+		 *
+		 * @param pattern The string wildcarded expression
+		 * @return a predicate whether we should direct to proxy
+		 */
+		public static RegexShouldProxyPredicate fromWildcardedPattern(String pattern) {
+			return new RegexShouldProxyPredicate(pattern.replaceAll("\\.", "\\\\.").replaceAll("\\*", "\\.*"));
+		}
+
+		private RegexShouldProxyPredicate(String pattern) {
+			this.regex = pattern;
+		}
+
+		private Pattern getPattern() {
+			if (pattern == null) {
+				pattern = Pattern.compile(regex, Pattern.CASE_INSENSITIVE);
+			}
+			return pattern;
+		}
+
+		/**
+		 * The test returns true when the given {@link SocketAddress} should be contacted through the configured proxy.
+		 * When this method returns false, the client will reach out to the address directly and bypass the proxy
+		 *
+		 * @param socketAddress
+		 * @return true when we should go through proxy
+		 */
+		@Override
+		public boolean test(SocketAddress socketAddress) {
```

Comment:
We have this `Builder nonProxyHostsPredicate(Predicate<SocketAddress> nonProxyHostsPredicate);`

```
		 * A standard predicate expression for a configured list of hosts that should be reached directly, bypassing
 		 * the proxy.
```

So the predicate returns true when you want to bypass the proxy while here it is the opposite?

Reply:
if that is the case I'd definitely recommend to make it consistent